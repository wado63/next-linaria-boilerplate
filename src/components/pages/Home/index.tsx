import { css, cx } from "linaria";
import type { FC, VFC } from "react";
import { fonts } from "../../../theme/fonts";
import { Button } from "../../uiKits/Button";
import { Background } from "./components/BackGround";

export const Home: VFC = () => (
  <main className={layout}>
    <Title className={layoutTitle} />
    <Description className={layoutDescription} />
    <CustomButton className={layoutButton} />
    <Background className={layoutBackGround} />
  </main>
);

const layout = css`
  display: grid;
  grid-template-columns: 1fr auto auto auto 1fr;
  grid-template-rows: 1fr auto 30px auto 30px auto 1fr;
  grid-template-areas:
    ". .           .           .           ."
    ". title       title       title       ."
    ". .           .           .           ."
    ". description description description ."
    ". .           .           .           ."
    ". button      .           .           ."
    ". .           .           .           .";

  min-height: 100vh;
`;

const layoutTitle = css`
  grid-area: title;
`;

const layoutDescription = css`
  grid-area: description;
`;

const layoutButton = css`
  grid-area: button;
`;

const layoutBackGround = css`
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

/**
 * Title Component
 */
const Title: FC<{ className: string }> = ({ className }) => (
  <h1 className={cx(className, styleTitle)}>Next-linaria-boilerplate</h1>
);

const styleTitle = css`
  font-family: ${fonts.family.FredokaOne};
  font-size: 5vw;
`;

/**
 * Description Component
 */
const Description: FC<{ className?: string }> = ({ className }) => (
  <p className={cx(className, styleDescription)}>
    This is an extension of &nbsp;
    <a
      href="https://github.com/vercel/next.js/tree/canary/examples/with-linaria"
      rel="noffolow"
    >
      &quot;the next with linaria example.&quot;
    </a>
  </p>
);

const styleDescription = css`
  font-size: 3vw;
`;

/**
 * CustomButton Component
 */
const CustomButton: FC<{ className?: string }> = ({ className }) => (
  <Button
    className={cx(className, styleButton)}
    onClick={() => {
      alert("hello world");
    }}
  >
    test button
  </Button>
);

const styleButton = css`
  background-color: skyblue;
  color: #000;
`;

export default Home;
