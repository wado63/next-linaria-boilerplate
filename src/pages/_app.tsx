import { AppPropsType } from "next/dist/next-server/lib/utils";
import { ReactElement } from "react";
import { GlobalStyles } from "../theme/components/GlobalStyles";

export default function MyApp({
  Component,
  pageProps,
}: AppPropsType): ReactElement {
  return (
    <>
      <GlobalStyles />
      <Component {...pageProps} />
    </>
  );
}
