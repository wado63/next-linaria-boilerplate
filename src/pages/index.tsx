import Head from "next/head";
import { Home as Page } from "../components/pages/Home";

export const Home = (): JSX.Element => (
  <div className="container">
    <Head>
      <title>With Linaria</title>
      <link rel="icon" href="/favicon.ico" />
    </Head>

    <Page />
  </div>
);

export default Home;
