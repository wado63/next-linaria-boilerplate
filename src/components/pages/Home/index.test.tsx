import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { Home } from "../../../pages/index";

describe("Home page", () => {
  it("should matches snapshot", () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Test Button", () => {
  it("should invoke an alert", () => {
    const alert = jest.fn();
    window.alert = alert;
    const { getByRole } = render(<Home />);
    fireEvent.click(getByRole("button"));
    expect(alert).toBeCalledTimes(1);
  });
});
