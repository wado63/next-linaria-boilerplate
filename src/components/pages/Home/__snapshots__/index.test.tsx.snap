// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="l1epjkbw"
    >
      <h1
        class="l18osnbr shfdegq"
      >
        Next-linaria-boilerplate
      </h1>
      <p
        class="l1qjyihy s1jpyom"
      >
        This is an extension of Â 
        <a
          href="https://github.com/vercel/next.js/tree/canary/examples/with-linaria"
          rel="noffolow"
        >
          "the next with linaria example."
        </a>
      </p>
      <button
        class="lbro7hf sofasun s3ww1ls"
      >
        test button
      </button>
      <div
        class="l1e9xby8 lyoxftv"
      >
        <div
          class="l1uaf86i"
        >
          <div
            class="s7qlbor s1y5i7kz"
          />
          <div
            class="p1x6vsh pszu3aq"
          />
          <div
            class="y18buyzd y1x93k46"
          />
          <div
            class="p1re9g4v pn4r4n7"
          />
          <div
            class="gha9cgk g1gkoc7f"
          />
        </div>
        <div
          class="l1uaf86i"
        >
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
          <div
            class="dix8bm0"
          />
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
